drizzlepac/hlautils/product.py

```diff
@@ -0,0 +1,222 @@
+        self.obset_id = obset_id
+        self.instrument = instrument
+        self.detector = detector
+
+        self.basename = "hst_" + "_".join(map(str, [prop_id, obset_id, instrument, detector]))
+
+        # exposure_name is the ipppssoo or a portion thereof
+        self.exposure_name = filename[0:8]
+        self.mjdutc = None
+
+        # class variables to be set later
+        self.param_dict = {}
+
+    def update_config(self, param_dict):
+        pass
+
+class TotalProduct(HAPProduct):
+    """ A Total Detection Product is a 'white' light mosaic comprised of
+        images acquired with one instrument, one detector, all filters, and all
+        exposure times
+
+        The "tdp" is short hand for TotalProduct.
+    """
+    def __init__(self, prop_id, obset_id, instrument, detector, filename):
+        super().__init__(prop_id, obset_id, instrument, detector, filename)
+
+        self.exposure_name = filename[0:6]
+
+        self.product_basename = self.basename + "_total_" + self.exposure_name
+        self.trl_filename = self.product_basename + "_trl.log"
+        self.point_cat_filename = self.product_basename + "_point-cat.ecsv"
+        self.segment_cat_filename = self.product_basename + "_segment-cat.ecsv"
+        self.drizzle_filename = ""
+
+        # How exactly is this used as it is set the number of times a TDP is created for
+        # an obset
+        # Manifest is created for the obset and instrument, regarding of the detector
+        self.manifest_name = '_'.join([instrument, filename[1:4], obset_id, "manifest.txt"])
+
+        # These attributes will be set later
+        self.edp_list = []
+        self.fdp_list = []
+        self.regions_dict = {}
+        self.meta_wcs = None
+
+    def add_member(self, edp):
+        self.edp_list.append(edp)
+
+    def add_product(self, fdp):
+        self.fdp_list.append(fdp)
+
+    def create_drizzle_filename(self):
+        self._create_drizzle_filename_helper(self.edp_list[0])
+        return self.drizzle_filename
+
+    def _create_drizzle_filename_helper(self, edp):
+        self.drizzle_filename = self.product_basename + "_" + edp.filetype + ".fits"
+
+    # There is a unique WCS for each TotalProduct product which is generated based upon
+    # the merging of all the ExposureProductss which comprise the specific TotalProduct.
+    def build_metawcs(self):
+        """
+        image_list = [element.full_filename for element in edp_list]
+        meta_wcs = wcs_functions.make_mosaic_wcs(image_list)
+        """
+        pass
+
+    def drizzle_product(self, meta_wcs, configobj):
+        """
+        Create the drizzle-combined total image using the meta_wcs as the reference output
+        """
+        edp_filenames = [element.full_filename for element in self.edp_list]
+        astrodrizzle.AstroDrizzle(input=edp_filenames,
+                                  output=self.drizzle_filename,
+                                  final_refimage=meta_wcs,
```

Comment:
Remember to explicitly set 'final_outnx' and 'final_outny' based on the values from the meta_wcs definition.

Reply:
Currently using a reference image instead of the meta_wcs.  However, I would like to use the meta_wcs instead.  This method has been updated to use the final_[outnx|outny].